addons/account/models/account_move.py

```diff
@@ -698,6 +698,20 @@ def _compute_base_line_taxes(base_line):
                 tax_line._onchange_amount_currency()
                 tax_line._onchange_balance()
 
+    def _tax_tags_need_inversion(self, is_refund, tax_type):
+        """ Tells whether the tax tags need to be inverted for this move under
+        the given parameters.
+
+        :param is_refund: whether or not the operation we want the inversion value for is a refund
+        :param tax_type: the tax type of the operation we want the inversion value for
+
+        :return: True if the tags need to be inverted
+        """
+        self.ensure_one()
```

Comment:
Either you don't give parameters if this is to be used as an instance method (with `self.ensure_one()`), either you provide the arguments but remove the `ensure_one`.
The first solution is the best IMO, even more so because of the way it is inherited.
Also, it would make more sense to be a function of `account.move.line`, called on the base line.

By the way, `tax_type` is unused in `_compute_base_line_taxes` when `self.type != 'entry'`.

Reply:
I removed the ensure_one and now pass the move type as parameter. I prefer to keep it there for readability; it's just a private function serving as a hook to slighlty change the behavior of the bigger one above. 

Anyway, this fix will only serve in 13-14. For versions above, the tax_tag_invert field already supports this case (I tested)

Reply:
That's exactly why we had like 10 successive bugfixes with the Swiss QR code.

Also, you can't call `self.id` if you remove the `self.ensure_one`

Reply:
I fail to see the connection with Swiss QR. What caused nearly all bugfixes there was the general confusion in the spec, and the number of successive corner cases that were reported with QR-IBANs and QRR that we did not cover.

Fixed the .id stuff.

Reply:
Most of the fixes were because the signature of the function kept on changing in all versions (literally all single version/saas is different). The values provided all depend on `self` and can be computed easily. No need to make it harder.

I really don't like this version and I am convinced that it is a really bad practice. But ok, I won't change your mind.

And why is the function on  `account.move`, when it is the line that needs or doesn't need a tag inversion?